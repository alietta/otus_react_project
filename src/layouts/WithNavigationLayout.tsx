import React, { useContext } from "react";
import { Container, useTheme } from "sancho";
import { Headbar } from "components/Headbar";

export const WithNavigationLayout: React.FC = (props) => {
  const theme = useTheme();
  const onLogout = () => {
    console.log('logout')
  };
  return (
    <div
      css={{
        width: "100vw",
        height: "100vh",
        background: theme.colors.background.default,
      }}
    >
      <Headbar onLogout={onLogout} />
      <Container css={{ maxWidth: 1024, paddingTop: 50 }}>
        {props.children}
      </Container>
    </div>
  );
};
