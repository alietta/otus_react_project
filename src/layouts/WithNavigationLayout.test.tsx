import React from "react";
import { mount } from "enzyme";
import { WithNavigationLayout } from "./WithNavigationLayout";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";

const mockStore = configureMockStore([thunk]);
const store = mockStore({
  user: {
    name: "Helen",
  },
});
describe("WithNavigationLayout", () => {
  const Component: React.FC<> = () => <h1>Test</h1>;
  it("render", () => {
    const element = mount(
      <WithNavigationLayout>
        <Component />
      </WithNavigationLayout>,
      {
        wrappingComponent: Provider,
        wrappingComponentProps: {
          store,
        },
      }
    );
    expect(element.html()).toMatchSnapshot();
  });
});
