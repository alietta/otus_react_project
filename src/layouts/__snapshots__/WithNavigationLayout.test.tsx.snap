// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithNavigationLayout render 1`] = `"<div class=\\"css-1x4340x-WithNavigationLayout\\"><div class=\\"css-94ghjt-Headbar\\"><div class=\\"css-hkw5ma-Headbar\\"><span class=\\"css-69rko3-Headbar\\">Helen</span><div class=\\"logout-icon css-1vhmik5-Headbar\\"><svg viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"#212529\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" height=\\"20px\\" width=\\"20px\\"><path d=\\"M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4\\"></path><polyline points=\\"16 17 21 12 16 7\\"></polyline><line x1=\\"21\\" y1=\\"12\\" x2=\\"9\\" y2=\\"12\\"></line></svg></div></div></div><div class=\\"css-u9wk6k-WithNavigationLayout\\"><h1>Test</h1></div></div>"`;
